ste/sender-blockBlob.go

```diff
@@ -273,6 +273,32 @@ func (s *blockBlobSenderBase) Cleanup() {
 	}
 }
 
+//Currently we've common Metadata Copier across all senders for block blob.
+func (s *blockBlobSenderBase) GenerateCopyMetadata(id common.ChunkID) chunkFunc {
+	return createChunkFunc(true, s.jptm, id, func() {
+		if s.jptm.Info().PreservePOSIXProperties {
+
+			if unixSIP, ok := s.sip.(IUNIXPropertyBearingSourceInfoProvider); ok {
+				// Clone the metadata before we write to it, we shouldn't be writing to the same metadata as every other blob.
+				s.metadataToApply = common.Metadata(s.metadataToApply).Clone().ToAzBlobMetadata()
```

Comment:
❗ If we're only going to persist POSIX properties, this would overwrite other unrelated metadata.

Reply:
The ask was to update all metadata. I think I kind of messed up the pr name.